src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,553 @@
+
+    @property
+    def si_value(self):
+        return self._unit.si_factor * self._value
+
+    @property
+    def si_unit(self):
+        return self._unit.si_unit
+
+    def is_dimension_less(self):
+        return all([value == 0 for value in self.get_dimensions_list()])
+
+    def get_dimensions_list(self):
+        dims = self._dimension.get_dimensions_dict()
+        return [
+            dims[_UnitsTable.dimension_order[key]]
+            for key in _UnitsTable.dimension_order.keys()
+        ]
+
+    def to(self, to_unit_str):
+        temp_quantity = Quantity(1, to_unit_str)
+
+        curr_unit_dim_obj = self.dimension
+        to_unit_dim_obj = temp_quantity.dimension
+
+        if (
+            curr_unit_dim_obj.get_dimensions_dict()
+            != to_unit_dim_obj.get_dimensions_dict()
+        ):
+            raise ValueError(
+                f"Incompatible conversion from {self.unit} : to {to_unit_str}"
+            )
+        else:
+            pass
+
+        curr_unit_obj = self._unit
+        to_unit_obj = temp_quantity._unit
+        temp_quantity.value = (
+            curr_unit_obj.si_factor / to_unit_obj.si_factor
+        ) * self.value
+
+        return temp_quantity
+
+    def _get_si_unit(self, other, func):
+        curr_dim = self.get_dimensions_list()
+        other_dim = other.get_dimensions_list()
+        temp_dim = [func(curr_dim[i], other_dim[i]) for i in range(len(other_dim))]
+        temp_unit = get_si_unit_from_dim(temp_dim)
+        return temp_unit
+
+    def __pow__(self, exponent):
+        new_dims = list(
+            map(lambda x: x * exponent if x != 0 else x, self.get_dimensions_list())
+        )
+        new_si_unit = get_si_unit_from_dim(new_dims)
+        new_si_value = pow(self.si_value, exponent)
+        return Quantity(new_si_value, new_si_unit)
+
+    def __float__(self):
+        return float(self.si_value)
+
+    def __str__(self):
+        return f'({self.value}, "{self.unit}")'
+
+    def __repr__(self):
+        return f'(Quantity ({self.value}, "{self.unit}"))'
+
+    def __mul__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self.si_value * other.si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x + y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = Quantity(self.si_value * other, self.si_unit)
+            return temp
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self.si_value / other.si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x - y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = Quantity(self.si_value / other, self.si_unit)
+            return temp
+
+    def __add__(self, other):
```

Comment:
Why is this working?
``` python
>>> a
(Quantity (0.5, "cm s^-1"))
>>> v
(Quantity (0.005, "m"))
>>> v + a
(Quantity (0.01, "m"))
>>>
```

Reply:
Now we are avoiding `+ , -` between incompatible dimensions. Thank you.